File: Python/ADMM_CNJGL/soft_scal_nodiag.py
# Soft-scaling operator with diagonals
# Not penalized

from numpy import zeros, diag, matrix
from soft_scal import soft_scal
from diag_construction import diag_construction
from nodiag_construction import nodiag_construction

def soft_scal_nodiag(Y, lam):
    p = Y.shape[1]
    X = matrix(zeros((p,p)))
    xtilde = matrix(zeros((p-1,p-1)))
    ytilde = nodiag_construction(Y)
    
    xtilde = soft_scal(ytilde, lam)
    return diag_construction(xtilde, diag(Y))

File: RLang/ADMM_CNJGL/soft_scal_nodiag.R
soft_scal_nodiag <- function(Y,lambda) {
  p <- dim(Y)[2];
  X <- matrix(0,p,p);
  Xtilde = matrix(0,p-1,p-1);
  Ytilde = nodiag_construction(Y);
  Xtilde = soft_scal(Ytilde,lambda);
  X = diag_construction(Xtilde,diag(Y));
}
